### ISSUE TITLE
Negative dice loss

### ISSUE THREAD

**Issue by kimihailv** [qubvel/segmentation_models#177]
Dataset loading and preprocessing:
```
img = cv2.imread(path.join(self.img_dir, self.files[i] + self.img_ext))
img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
mask = cv2.imread(path.join(self.mask_dir, self.files[i] + self.mask_ext), 0)
mask = to_categorical(mask, num_classes=self.num_classes)

if self.augmentation is not None:
   augumented = self.augmentation(image=img, mask=mask)
   img, mask = augumented["image"], augumented["mask"]

if self.preprocessing is not None:
    sample = self.preprocessing(image=img, mask=mask)
    img, mask = sample["image"], sample["mask"]

return img, mask
```
Preprocessing function:
```
def get_preprocessing(preprocessing_fn):
    transform = [
        A.Lambda(image=preprocessing_fn),
    ]
 return A.Compose(transform)
```
Where preprocessing_fn is sm.get_preprocessing(backbone).
Where is my mistake? Thanks for the help.

**Comment by qubvel** [comment]
check than your mask is in range 0..1

**Comment by kimihailv** [comment]
I have already checked, and yes, values are in [0, 1]. I make np.clip(0, 1) in augmentation

**Comment by kimihailv** [comment]
> check that your mask is in range 0..1

`to_categorical` function in my code is `tf.keras.to_categorical` which covert a matrix with shape (h, w) to categorical the matrix with shape (h, w, num_classes)
